File: 258-AddDigits/258-AddDigits.py
# Last updated: 4/3/2026, 2:00:52 PM
class Solution(object):
    def addDigits(self, num):
        """
        :type num: int
        :rtype: int
        """
        if num==0:
            return 0
        elif num%9==0:
            return 9
        else:
            return num%9

File: 258-AddDigits/258-AddDigits.cpp
// Last updated: 4/3/2026, 2:00:57 PM
class Solution {
public:
    int addDigits(int num) {
        int sum=0;
        while(num>=10){
            sum=0;
            while(num!=0){
                sum+=num%10;
                num/=10;
            }
            num=sum;
        }
        return num;
    }
};